src/codemagic/tools/app_store_connect/action_groups/devices_action_group.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+from abc import ABCMeta
+from typing import List
+from typing import Optional
+from typing import Sequence
+from typing import Union
+
+from codemagic import cli
+from codemagic.apple.resources import BundleIdPlatform
+from codemagic.apple.resources import Device
+from codemagic.apple.resources import DeviceStatus
+from codemagic.cli import Colors
+
+from ..abstract_base_action import AbstractBaseAction
+from ..action_group import AppStoreConnectActionGroup
+from ..arguments import BundleIdArgument
+from ..arguments import DeviceArgument
+from ..arguments import Types
+from ..errors import AppStoreConnectError
+
+
+class DevicesActionGroup(AbstractBaseAction, metaclass=ABCMeta):
+    @cli.action(
+        "list",
+        BundleIdArgument.PLATFORM_OPTIONAL,
+        DeviceArgument.DEVICE_NAME_OPTIONAL,
+        DeviceArgument.DEVICE_STATUS,
+        action_group=AppStoreConnectActionGroup.DEVICES,
+        deprecation_info=cli.ActionDeprecationInfo("list-devices", "0.49.0"),
+    )
+    def list_devices(
+        self,
+        platform: Optional[BundleIdPlatform] = None,
+        device_name: Optional[str] = None,
+        device_status: Optional[DeviceStatus] = None,
+        should_print: bool = True,
+    ) -> List[Device]:
+        """
+        List Devices from Apple Developer portal matching given constraints
+        """
+
+        device_filter = self.api_client.devices.Filter(
+            name=device_name,
+            platform=platform,
+            status=device_status,
+        )
+        return self._list_resources(device_filter, self.api_client.devices, should_print)
+
+    @cli.action(
+        "register",
+        BundleIdArgument.PLATFORM,
+        DeviceArgument.DEVICE_NAME,
+        DeviceArgument.DEVICE_UDIDS,
+        DeviceArgument.IGNORE_REGISTRATION_ERRORS,
+        action_group=AppStoreConnectActionGroup.DEVICES,
+        deprecation_info=cli.ActionDeprecationInfo("register-device", "0.49.0"),
+    )
+    def register_device(
```

Comment:
Used to be defined in https://github.com/codemagic-ci-cd/cli-tools/blob/3670fa624997f17cb437bc88198f7bdee3594f35/src/codemagic/tools/app_store_connect.py#L323-L385